src/NLog/Internal/XmlHelper.cs

```diff
@@ -58,7 +59,33 @@ public static class XmlHelper
         /// </summary>
         private static string RemoveInvalidXmlChars(string text)
         {
-            return String.IsNullOrEmpty(text) ? "" : InvalidXmlChars.Replace(text, "");
+            if (string.IsNullOrEmpty(text))
+                return string.Empty;
+
+#if NET3_5
+            return InvalidXmlChars.Replace(text, "");
+#else
+            StringBuilder sb = null;
+            for (int i = 0; i < text.Length; ++i)
+            {
+                char ch = text[i];
+                if (XmlConvert.IsXmlChar(ch))
+                {
+                    if (sb != null)
+                        sb.Append(ch);
```

Comment:
Also dunno if this performance wise OK. In the best case it is, but not in the worst case, as adding all chars to a stringBuilder in a for loop is less performant then substring.